### PR TITLE
model: added Querit/Querit

### PR DESCRIPTION
**Pull request by youngbeauty250**
- [x] I have filled out the ModelMeta object to the extent possible
- [x] I have ensured that my model can be loaded using
  - [x] `mteb.get_model(model_name, revision)` and
  - [x] `mteb.get_model_meta(model_name, revision)`
- [x] I have tested the implementation works on a representative set of tasks.
- [x] The model is public, i.e., is available either as an API or the weights are publicly available to download

### CODE REVIEW

File: mteb/models/model_implementations/querit_models.py [hunk 1]
```diff
@@ -0,0 +1,401 @@
...
+from tqdm.auto import tqdm
+from transformers import (
+    AutoTokenizer,
+    DeepseekV3ForCausalLM,
+)
+
+
+logger = logging.getLogger(__name__)
+
+class QueritModel(DeepseekV3ForCausalLM):
```
Comment (Samoed) [inline mteb/models/model_implementations/querit_models.py]: Can you move this to your model implementation to make your model usable with `trust_remote_code`?

Comment (youngbeauty250) [reply mteb/models/model_implementations/querit_models.py]: I have moved it

File: mteb/models/model_implementations/querit_models.py [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+
+class QueritWrapper(RerankerWrapper):
+    """
+    Multi-GPU / multi-process reranker wrapper for mteb.mteb evaluation.
+    Supports flattening all query-passage pairs without explicit grouping.
+    """
+
+    def __init__(
+        self,
+        model_name_or_path: str,
```
Comment (Samoed) [inline mteb/models/model_implementations/querit_models.py]:
```suggestion
        model_name: str,
        revision: str,
        device: str,
```

File: mteb/models/model_implementations/querit_models.py [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+        self.tokenizer = AutoTokenizer.from_pretrained(
+            model_name_or_path, trust_remote_code=True
+        )
+        if "[CLS]" not in self.tokenizer.get_vocab():
+            raise ValueError("Tokenizer missing required special token '[CLS]'")
+        self.cls_token_id = self.tokenizer.convert_tokens_to_ids("[CLS]")
+        self.pad_token_id = self.tokenizer.pad_token_id or 0
+
+        self.max_length = min(
+            kwargs.get("max_length", 4096), self.tokenizer.model_max_length
```
Comment (Samoed) [inline mteb/models/model_implementations/querit_models.py]: Is this expected to be used up to 4096 tokens or you support 131k?

File: mteb/models/model_implementations/querit_models.py [hunk 4]
```diff
@@ -0,0 +1,241 @@
...
+        - Left-pad to max_length
+        - Generate custom attention mask based on block types
+        """
+        max_length = self.max_length
+        # Construct input texts
+        enc = self.tokenizer(
+            pairs,
+            add_special_tokens=False,
+            truncation=True,
+            max_length=max_length - 1,
```
Comment (Samoed) [inline mteb/models/model_implementations/querit_models.py]: Maybe this better to do in `__init__`?

File: mteb/models/model_implementations/querit_models.py [hunk 5]
```diff
@@ -0,0 +1,241 @@
...
+                out = self.model(**enc)
+                scores = out["score"].squeeze(-1).detach().float().cpu().tolist()
+
+                if not isinstance(scores, list):
+                    scores = [scores]
+
+                final_scores.extend(scores)
+                pbar.update(len(scores))
+
+        assert len(final_scores) == num_pairs
```
Comment (Samoed) [inline mteb/models/model_implementations/querit_models.py]: Can you remove asserts?
```suggestion
```

File: mteb/models/model_implementations/querit_models.py [hunk 6]
```diff
@@ -0,0 +1,241 @@
...
+        type_mask = mask_content | mask_cls
+        return type_mask.unsqueeze(0)
+
+
+model_meta = ModelMeta(
+    loader=QueritWrapper,
+    loader_kwargs={
+        "fp_options": "bfloat16",
+    },
+    name="Querit/Querit",
```
Comment (Samoed) [inline mteb/models/model_implementations/querit_models.py:225]: Can you create model card in your repository for your model?

File: mteb/models/model_implementations/querit_models.py [hunk 7]
```diff
@@ -0,0 +1,241 @@
...
+    model_type=["cross-encoder"],
+    languages=["eng"],
```
Comment (Samoed) [inline mteb/models/model_implementations/querit_models.py]:
```suggestion
    languages=["eng-Latn"],
```

File: mteb/models/model_implementations/querit_models.py [hunk 8]
```diff
@@ -0,0 +1,241 @@
...
+    open_weights=True,
+    revision="5ad2649cc4defb7e1361262260e9a781f14b08bc",
+    release_date="2026-01-24",
+    n_parameters=None,
```
Comment (Samoed) [inline mteb/models/model_implementations/querit_models.py]: Can you fill this?

File: mteb/models/model_implementations/querit_models.py [hunk 9]
```diff
@@ -0,0 +1,241 @@
...
+    n_embedding_parameters=None,
+    memory_usage_mb=9383.0,
+    max_tokens=None,
```
Comment (Samoed) [inline mteb/models/model_implementations/querit_models.py]: Can you fill this?

File: mteb/models/model_implementations/querit_models.py [hunk 10]
```diff
@@ -0,0 +1,241 @@
...
+            self.device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+        model_args = {}
+        if self.fp_options:
+            model_args["torch_dtype"] = self.fp_options
+        self.model = AutoModel.from_pretrained(
+            model_name_or_path, trust_remote_code=True, **model_args
+        )
+        logger.info(f"Using model {model_name_or_path}")
+
+        if kwargs.get("torch_compile"):
```
Comment (Samoed) [inline mteb/models/model_implementations/querit_models.py]: Can you add this to `__init__` params?

File: mteb/models/model_implementations/querit_models.py [hunk 11]
```diff
@@ -0,0 +1,241 @@
...
+            self.torch_compile = kwargs["torch_compile"]
```
Comment (Samoed) [inline mteb/models/model_implementations/querit_models.py]: `self.torch_compile` is not used

File: mteb/models/model_implementations/querit_models.py [hunk 12]
```diff
@@ -0,0 +1,241 @@
...
+                final_scores.extend(scores)
+                pbar.update(len(scores))
+
+        assert len(final_scores) == num_pairs
+        return final_scores
+
+    @staticmethod
+    def format_instruction(instruction: str | None, query: str, doc: str) -> str:
+        if instruction is None:
+            output = f"Judge whether the Content meets the requirements based on the Query. Query: {query}; Content: {doc}"
```
Comment (Samoed) [inline mteb/models/model_implementations/querit_models.py:167]: Is it correct that this don't require `<Instruct>`?

File: mteb/models/model_implementations/querit_models.py [hunk 13]
```diff
@@ -0,0 +1,241 @@
...
+        if "instruction" in inputs2.dataset.features:
+            instructions = [text for batch in inputs1 for text in batch["instruction"]]
+
+        num_pairs = len(queries)
+        if num_pairs == 0:
+            return []
+        final_scores: list[float] = []
+
+        with tqdm(total=num_pairs, desc="Scoring", ncols=100) as pbar:
+            for start in range(0, num_pairs, self.batch_size):
```
Comment (Samoed) [inline mteb/models/model_implementations/querit_models.py]: It's better to use batch_size that passed to `predict` function (it will be in kwargs)

File: mteb/models/model_implementations/querit_models.py [hunk 14]
```diff
@@ -0,0 +1,241 @@
...
+                end = min(start + self.batch_size, num_pairs)
+                batch_q = queries[start:end]
+                batch_d = passages[start:end]
+
+                batch_instructions = (
```
Comment (Samoed) [inline mteb/models/model_implementations/querit_models.py:143]: You can also get task specific instruction with https://github.com/embeddings-benchmark/mteb/blob/7c9bab2f8ed4f3992c681098c232524eb6043621/mteb/models/abs_encoder.py#L216

File: mteb/models/model_implementations/querit_models.py (hunk 2 above)
Comment (youngbeauty250) [reply mteb/models/model_implementations/querit_models.py]: OK, I will change it

File: mteb/models/model_implementations/querit_models.py (hunk 5 above)
Comment (youngbeauty250) [reply mteb/models/model_implementations/querit_models.py]: I have moved it

File: mteb/models/model_implementations/querit_models.py (hunk 6 above)
Comment (youngbeauty250) [reply mteb/models/model_implementations/querit_models.py:225]: We will update the model card as soon as possible in the coming days.

File: mteb/models/model_implementations/querit_models.py (hunk 4 above)
Comment (youngbeauty250) [reply mteb/models/model_implementations/querit_models.py]: I have moved it.

File: mteb/models/model_implementations/querit_models.py (hunk 3 above)
Comment (youngbeauty250) [reply mteb/models/model_implementations/querit_models.py]: Our tokenizer supports up to 131k tokens, but most of our training data does not exceed 40,96 tokens, so we expect the maximum token length to be 40,96.

File: mteb/models/model_implementations/querit_models.py (hunk 14 above)
Comment (youngbeauty250) [reply mteb/models/model_implementations/querit_models.py:143]: Thank you for your suggestion. We will refer to this code to get task specific instruction.

File: mteb/models/model_implementations/querit_models.py (hunk 13 above)
Comment (youngbeauty250) [reply mteb/models/model_implementations/querit_models.py]: I have changed to it.

File: mteb/models/model_implementations/querit_models.py (hunk 12 above)
Comment (youngbeauty250) [reply mteb/models/model_implementations/querit_models.py:167]: We follow the format of other reranker

File: mteb/models/model_implementations/querit_models.py (hunk 11 above)
Comment (youngbeauty250) [reply mteb/models/model_implementations/querit_models.py]: I have moved it

File: mteb/models/model_implementations/querit_models.py (hunk 10 above)
Comment (youngbeauty250) [reply mteb/models/model_implementations/querit_models.py]: I have moved it

File: mteb/models/model_implementations/querit_models.py (hunk 9 above)
Comment (youngbeauty250) [reply mteb/models/model_implementations/querit_models.py]: I will fill it

File: mteb/models/model_implementations/querit_models.py (hunk 7 above)
Comment (youngbeauty250) [reply mteb/models/model_implementations/querit_models.py]: Thanks. I will change it.

File: mteb/models/model_implementations/querit_models.py (hunk 8 above)
Comment (youngbeauty250) [reply mteb/models/model_implementations/querit_models.py]: I will fill it.

File: mteb/models/model_implementations/querit_models.py (hunk 12 above)
Comment (Samoed) [reply mteb/models/model_implementations/querit_models.py:167]: But this is not matching with your reranker I think

Comment (youngbeauty250) [reply mteb/models/model_implementations/querit_models.py:167]: emmmmmmm.....We follow the instruction of Qwen3-Reranker. Maybe it exists a better instruction. We will think more about it.

Comment (Samoed) [reply mteb/models/model_implementations/querit_models.py:167]: I see this, but I think your templates should be consistent either use `<Query>: ` or `Query: `. Also you don't include here chat roles which can degrade performance `<|im_start|>system\n` probably

Comment (youngbeauty250) [reply mteb/models/model_implementations/querit_models.py:167]: Thank you. We have align the templates with our training templates. For the chat roles, we will seriously consider your suggestions and make improvements in future versions.

Comment (Samoed) [reply mteb/models/model_implementations/querit_models.py:167]: You mentioned that you’re following the Qwen3-Reranker instruction, I thought those were applied to the roles as well

File: mteb/models/model_implementations/querit_models.py [hunk 15]
```diff
@@ -0,0 +1,233 @@
...
+        "fp_options": "bfloat16",
+    },
+    name="Querit/Querit",
+    model_type=["cross-encoder"],
+    languages=["eng-Latn"],
+    open_weights=True,
+    revision="5ad2649cc4defb7e1361262260e9a781f14b08bc",
+    release_date="2026-01-24",
+    n_parameters=4919636992,
+    n_embedding_parameters=1024,
```
Comment (Samoed) [inline mteb/models/model_implementations/querit_models.py]: I don't think that in your embedding parameters you have only 1024

File: mteb/models/model_implementations/querit_models.py [hunk 16]
```diff
@@ -0,0 +1,233 @@
...
+    memory_usage_mb=9383.0,
+    max_tokens=4096,
+    reference="https://huggingface.co/Querit/Querit",
+    similarity_fn_name=None,
+    training_datasets=set(),
```
Comment (Samoed) [inline mteb/models/model_implementations/querit_models.py]: Can you add your training datasets?

File: mteb/models/model_implementations/querit_models.py [hunk 17]
```diff
@@ -0,0 +1,233 @@
...
+    embed_dim=None,
+    license="apache-2.0",
+    framework=["PyTorch"],
+    use_instructions=None,
+    public_training_code=None,
+    public_training_data=None,
+    citation=None,
```
Comment (Samoed) [inline mteb/models/model_implementations/querit_models.py:244]: Can you fill `adapted_from` with your source model?

Comment (youngbeauty250) [reply mteb/models/model_implementations/querit_models.py:244]: We don't fill in "adapted_from" because this model was trained completely from scratch.  
We just followed and changed the DeepSeek-V3 architecture , but all weights are our own — no continued pre-training or fine-tuning from their checkpoint. Even in this case, should we still fill in `adapted_from`？

File: mteb/models/model_implementations/querit_models.py (hunk 15 above)
Comment (youngbeauty250) [reply mteb/models/model_implementations/querit_models.py]: I have revised it. The `embed_dim` is 1024.

File: mteb/models/model_implementations/querit_models.py (hunk 16 above)
Comment (youngbeauty250) [reply mteb/models/model_implementations/querit_models.py]: I have added it. Thanks for your suggestions overall ^_^

File: mteb/models/model_implementations/querit_models.py (hunk 17 above)
Comment (Samoed) [reply mteb/models/model_implementations/querit_models.py:244]: In that case you don't need this. I thought you're using qwen3

File: mteb/models/model_implementations/querit_models.py (hunk 16 above)
Comment (Samoed) [reply mteb/models/model_implementations/querit_models.py]: Yes, but you don't list there anything? Will you share a list of your training datasets or is it fully synthetic/private?

File: mteb/models/model_implementations/querit_models.py (hunk 15 above)
Comment (Samoed) [reply mteb/models/model_implementations/querit_models.py]: We have separate parameter for embed dim https://github.com/embeddings-benchmark/mteb/blob/47d59ad56b36b6b4be2a6cdc8b0a092161e33ea3/mteb/models/model_meta.py#L111

This is number of how much embedder layer takes from model https://github.com/embeddings-benchmark/mteb/blob/47d59ad56b36b6b4be2a6cdc8b0a092161e33ea3/mteb/models/model_meta.py#L106

You can run this with 
```python
import numpy as np
from transformers import AutoModel

model = AutoModel.from_pretrained(model_name)

input_params = np.prod(model.get_input_embeddings().weight.shape)
```

File: mteb/models/model_implementations/querit_models.py (hunk 16 above)
Comment (Samoed) [reply mteb/models/model_implementations/querit_models.py]: Found in new commit

File: mteb/models/model_implementations/querit_models.py [hunk 18]
```diff
@@ -0,0 +1,244 @@
...
+    "MIRACLRanking",  # https://huggingface.co/datasets/mteb/MIRACLReranking
+    "mrtidy",  # https://huggingface.co/datasets/mteb/mrtidy
+    "ruri-v3-dataset-reranker",  # https://huggingface.co/datasets/cl-nagoya/ruri-v3-dataset-reranker
+    "MLDR",  # https://huggingface.co/datasets/Shitao/MLDR
+    "MindSmallReranking",  # https://huggingface.co/datasets/mteb/MindSmallReranking
+    "MSMARCO",  # https://huggingface.co/datasets/mteb/msmarco
+    "CQADupStack",  # https://huggingface.co/datasets/mteb/cqadupstack-*
+    "AskUbuntu",
+    "T2Reranking",
+}
```
Comment (Samoed) [inline mteb/models/model_implementations/querit_models.py:218]:
```suggestion
querit_reranker_training_data = {
    "MIRACLRanking",  # https://huggingface.co/datasets/mteb/MIRACLReranking
    "MrTidyRetrieval",  # https://huggingface.co/datasets/mteb/mrtidy
    "ruri-v3-dataset-reranker",  # https://huggingface.co/datasets/cl-nagoya/ruri-v3-dataset-reranker
    "MultiLongDocReranking",  # https://huggingface.co/datasets/Shitao/MLDR
    "MindSmallReranking",  # https://huggingface.co/datasets/mteb/MindSmallReranking
    "MSMARCO",  # https://huggingface.co/datasets/mteb/msmarco
    "CQADupStack",  # https://huggingface.co/datasets/mteb/cqadupstack-*
    "AskUbuntuDupQuestions",
    "T2Reranking",
}
```

File: mteb/models/model_implementations/querit_models.py (hunk 15 above)
Comment (youngbeauty250) [reply mteb/models/model_implementations/querit_models.py]: I have added the `n_embedding_parameters`

File: mteb/models/model_implementations/querit_models.py (hunk 18 above)
Comment (youngbeauty250) [reply mteb/models/model_implementations/querit_models.py:218]: I have revised it
